qcengine/util.py

```diff
@@ -377,6 +377,7 @@ def execute(
     infiles: Optional[Dict[str, str]] = None,
     outfiles: Optional[List[str]] = None,
     *,
+    outfiles_track: Optional[List[str]] = [],
```

Comment:
Probably better to have `None` here, like `as_binary`. https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
Good point but shouldn't matter here since `outfiles_track` should never be modified, but if you're concerned someone might I could change its type to an immutable object e.g.  
```python
outfiles_track: Optional[Tuple[str]] = (),
```
If you prefer a list for consistency that is by default `None`, I could add a single line to `disk_files`:
```python
outfiles_track = outfiles_track or []
```
that would make the current code work with no additional changes.

Reply:
Good point but shouldn't matter here since `outfiles_track` should never be modified, but if you're concerned someone might I could change its type to an immutable object like a `Tuple` OR if you prefer a list for consistency that is by default `None`, I could add a single line to `disk_files`:

```python
outfiles_track = outfiles_track or []
```
that would make the current code work with no additional changes.